src/pydjinni/generator/cpp/cpp/templates/source/enum.cpp.jinja2

```diff
@@ -0,0 +1,34 @@
+{% extends "base.jinja2" %}
+{% from "macros.jinja2" import to_string %}
+
+{% block header %}
+#include {{ type_def.cpp.header | quote }}
+{% if config.string_serialization_for_enums %}
+#include <string>
+{% endif %}
+{% endblock %}
+
+{% block content %}
+inline std::string to_string( {{ type_def.cpp.name }} value)
+{
+    static const std::pair<{{ type_def.cpp.name }}, std::string> m[] = { 
+{% for item in type_def.items %}
+         { {{type_def.cpp.name}}::{{item.cpp.name}},"{{item.cpp.name}}"},
+{% endfor %}
+    };
+    auto it = std::find_if(std::begin(m), std::end(m),
```

Comment:
you don't need add `#include <algorithm>`?

Reply:
Another thing that im thinking here. The method name `to_string` will be fixed snake_case or it will follow the configured case of methods for C++ generation?

Reply:
This `to_string` was meant to be a private implementation method - I should put it in an anonymous namespace - so for me it is not very relevant what style is being used for public APIs. That said, it's easy enough to follow the configuration, I guess - but I prefer to keep the complexity of the template as low as possible.

Reply:
(I made it a private implementation method, instead of directly putting it in the << operator implementation, because I plan to re-use it for json serialization)

Reply:
Ah ok, it is private. How can i use it in a real scene? I always need use stream? 

I think that will be complex always need do:

```
auto flag_value = ExampleFlag::XYZ;
std::stringstream stringified;
stringified << flag_value;
std::string real_value = stringified.str();
```

Instead, i can't do it direct? Example:

```
auto flagValue = ExampleFlag::XYZ::to_string();
```

or 

```
auto flagValue = ExampleFlag::XYZ::toString();
```

Reply:
Hmm. I thought I'd stick with the ostream `<<` operator overload, but I'd be happy to have helpers, too. Though, as you say, it's hard to say how much flexibility should be supported.

Reply:
So, you cannot create something that convert to string direct? 

```
std::string value << ExampleFlag::XYZ;
```

or 

```
std::string value = ExampleFlag::XYZ;
```

It is possible?

Reply:
IMO this boils down to a discussion about how the whole C++ ostream concept feels a bit outdated and weird. But until `std::format` sees wider compiler support, ostream seems to be the best thing we have, so I'm happy with what's proposed in this PR. :)

Reply:
I'm happy too, I'm just trying to see if there is something that isn't so verbose when using this in practice.

Reply:
I learned rn that [`fmt::format`](https://fmt.dev/latest/syntax.html) also works with ostream `<<` overloads, so you could write sth like `std::string value = fmt::format("{}", flag_value)`.
Unfortunately it seems like in MSVCs `std::format` did [not adopt this feature](https://devblogs.microsoft.com/cppblog/format-in-visual-studio-2019-version-16-10/), so in the long term special `std::formatter` serializers for records and also enums will be required in pydjinni...